### ISSUE TITLE
Loop in: Invalid swap invoice amount

### ISSUE THREAD

**Issue by hieblmi** [lightninglabs/loop#566]
Loop-ins with amounts > 1M sat temporarily failed to broadcast due to the following error:

```[ERR] LOOPD: Loop in: rpc error: code = InvalidArgument desc = cannot initiate swap: invalid swap invoice amount```

Trying it a few hours later this error disappeared.

**Comment by bhandras** [comment]
Thanks for reporting @hieblmi! This kind of error may happen because the probed fee is different from the actual routing fee. We're executing a probe to the target node to make sure it is reachable and also to give a reasonable fee estimate but sometimes route conditions change in the short time while the swap is executed and the swap invoice may be be out of date by the the server attempts to pay.

**Comment by hieblmi** [comment]
Thanks for the explanation @bhandras.

I tried looping in 0.4 BTC to a direct channel with LOOP which resulted in the error above. The probe and swap were executed with little delay in between. Hence I think a network change was unlikely the reason for the failure.
Right after this I tried looping in 0.01 BTC which succeeded. Here is the log of all that:
```
20:36:06.419 [INF] LOOPD: Loop in quote request received
20:36:09.839 [INF] LOOPD: Loop in request received
20:36:09.839 [INF] LOOP: Loop in 0.4 BTC (last hop: 021c97a90a411ff2b10dc2a8e32de2f29d2fa49d41bfbb52bd416e460db0747d0d)
20:36:10.280 [INF] LOOP: Creating probe invoice ec11269982f1f16bcb1da12f2a7bd331d7326ab1558dcbac001ecceb3e056ae9
20:36:10.351 [INF] LOOP: Initiating swap request at height 784811
20:36:11.267 [INF] LOOP: Server probe successful
20:36:12.303 [ERR] LOOPD: Loop in: rpc error: code = InvalidArgument desc = cannot initiate swap: invalid swap invoice amount
20:38:35.029 [INF] LOOPD: Loop in quote request received
20:38:37.588 [INF] LOOPD: Loop in request received
20:38:37.588 [INF] LOOP: Loop in 0.01 BTC (last hop: 021c97a90a411ff2b10dc2a8e32de2f29d2fa49d41bfbb52bd416e460db0747d0d)
20:38:38.053 [INF] LOOP: Creating probe invoice 03b8a7a9362218a1b4f5e70a1723a761041a2621fde17e2de05daeb38d1bc7cc
20:38:38.122 [INF] LOOP: Initiating swap request at height 784811
20:39:06.054 [INF] LOOP: Server probe successful
20:39:07.065 [INF] LOOP: 7698bc Htlc address (P2WSH): 
20:39:07.069 [INF] LOOP: 7698bc Server message: Use AutoLoop - it is really good, give it a try
20:39:07.069 [INF] LOOP: 7698bc Loop in swap state: Initiated
20:39:07.069 [INF] LOOP: 7698bc Blocks left until on-chain expiry: 1000
20:39:07.078 [INF] LOOP: 7698bc Loop in swap state: HtlcPublished
20:39:07.078 [INF] LOOP: 7698bc Publishing on chain HTLC with fee rate 3552 sat/kw
20:39:07.101 [INF] LOOP: 7698bc Published on chain HTLC tx , fee: 0.00002188 BTC
20:39:07.102 [INF] LOOP: 7698bc Register for htlc conf (hh=784811, txid=)
20:39:07.249 [INF] LOOP: 7698bc Server update: SERVER_INITIATED received, timestamp: 2023-04-10 20:39:06.982655713
```
